maven-core/src/main/java/org/apache/maven/execution/DefaultMavenExecutionRequest.java

```diff
@@ -687,6 +687,11 @@ public MavenExecutionRequest setPom( File pom )
     {
         this.pom = pom;
 
+        if ( this.pom != null && this.pom.getParentFile() != null )
```

Comment:
IIRC this will go wrong. There are a couple of plugins that want to rewrite the pom (maven-shade-plugin, flatten-maven-plugin). This pom will be in the target directory, so it can be cleaned up. However, the basedir should stay the same.

Reply:
Reverted that commit.